pkg/cloud/gen/main.go

```diff
@@ -886,17 +881,20 @@ func (g *{{.GCEWrapType}}) Insert(ctx context.Context, key *meta.Key, obj *{{.FQ
 		Service: "{{.Service}}",
 	}
 
-	klog.V(5).Infof("{{.GCEWrapType}}.Insert(%v, %v, ...): projectID = %v, ck = %+v", ctx, key, projectID, ck)
+	klog.V(5).Infof("{{.GCPWrapType}}.Insert(%v, %v, ...): projectID = %v, ck = %+v", ctx, key, projectID, ck)
 	callObserverStart(ctx, ck)
 	if err := g.s.RateLimiter.Accept(ctx, ck); err != nil {
-		klog.V(4).Infof("{{.GCEWrapType}}.Insert(%v, %v, ...): RateLimiter error: %v", ctx, key, err)
+		klog.V(4).Infof("{{.GCPWrapType}}.Insert(%v, %v, ...): RateLimiter error: %v", ctx, key, err)
 		return err
 	}
 	obj.Name = key.Name
 
 {{- if .IsNetworkServices}}
 	parent := fmt.Sprintf("projects/%s/locations/global", projectID)
 	call := g.s.{{.GroupVersionTitle}}.{{.Service}}.Create(parent, obj)
+	{{- if eq .Object "TcpRoute"}}
+	  call.TcpRouteId(obj.Name)
```

Comment:
Assuming this applies more broadly - do you know which other APIs would require something like this? If we leave this hardcoded in we may end up getting unnecessarily stuck debugging later whenever we add more resources like HTTPRoute.

Reply:
It seems all the routes require the ID. I've made the change to append the ID for all routes.